core/tests/ts-integration/tests/api/snapshots-creator.test.ts

```diff
@@ -0,0 +1,84 @@
+import { TestMaster } from '../../src/index';
+import fs from 'fs';
+import * as zlib from 'zlib';
+import { snapshots_creator } from 'zk/build/run/run';
+describe('Snapshots API tests', () => {
+    let testMaster: TestMaster;
+
+    beforeAll(() => {
+        testMaster = TestMaster.getInstance(__filename);
+
+        if (process.env.ZKSYNC_ENV!.startsWith('ext-node')) {
+            console.warn("You are trying to run snapshots creator tests on external node. It's not supported.");
+        }
+    });
+
+    async function runCreator() {
+        console.log('Starting creator');
+        await snapshots_creator();
+    }
+
+    async function rpcRequest(name: string, params: any) {
+        const response = await testMaster.mainAccount().provider.send(name, params);
+        console.log(response);
+        return response;
+    }
+
+    async function getAllSnapshots() {
+        return await rpcRequest('snapshots_getAllSnapshots', []);
+    }
+
+    async function getSnapshot(snapshotL1Batch: number) {
+        return rpcRequest('snapshots_getSnapshot', [snapshotL1Batch]);
+    }
+
+    async function decompressGzip(filePath: string): Promise<string> {
+        return new Promise((resolve, reject) => {
+            const readStream = fs.createReadStream(filePath);
+            const gunzip = zlib.createGunzip();
+            let data = '';
+
+            gunzip.on('data', (chunk) => (data += chunk.toString()));
+            gunzip.on('end', () => resolve(data));
+            gunzip.on('error', reject);
+
+            readStream.pipe(gunzip);
+        });
+    }
+    async function createAndValidateSnapshot() {
+        let existingBatchNumbers = (await getAllSnapshots()).snapshotsL1BatchNumbers as number[];
+        await runCreator();
+        let newBatchNumbers = (await getAllSnapshots()).snapshotsL1BatchNumbers as number[];
+        let addedSnapshots = newBatchNumbers.filter((x) => existingBatchNumbers.indexOf(x) === -1);
+        expect(addedSnapshots.length).toEqual(1);
+
+        let l1BatchNumber = addedSnapshots[0];
+        let fullSnapshot = await getSnapshot(l1BatchNumber);
+        let miniblockNumber = fullSnapshot.miniblockNumber;
+
+        expect(fullSnapshot.l1BatchNumber).toEqual(l1BatchNumber);
+        for (let chunkMetadata of fullSnapshot.storageLogsChunks) {
+            console.log(`Verifying ${chunkMetadata.filepath}`);
+            let path = `${process.env.ZKSYNC_HOME}/${chunkMetadata.filepath}`;
+
+            let output = JSON.parse(await decompressGzip(path));
+            expect(output['storageLogs'].length > 0);
```

Comment:
Dumb question: Doesn't this check contradict the fact the discussion about minimum chunk amount, in which you've said that some chunks may end up empty?

Reply:
this test implicitly depends on the fact that there are some storage logs (created during context setup), This assert it's a bit different, if it fails, that means that either one of those assumptions about the setup is no longer valid (and the test should be corrected as it may no longer check anything and fail silently) or that there is something very wrong with snapshots generation

Reply:
Aren't `output['storageLogs']` logs for a specific chunk? If so, it could be the case that we do have storage logs (e.g., ones in from the genesis miniblock), but some chunks don't have storage logs (IIRC, there's about 20 logs in the genesis miniblock). This isn't a blocking issue, just want to find out whether I understand test logic correctly.